runtime/cudaq/platform/mqpu/MultiQPUPlatform.cpp

```diff
@@ -9,100 +9,128 @@
-    cudaq::info("MultiQPUPlatform::resetExecutionContext QPU {}", qpu_id);
-    auto tid = std::hash<std::thread::id>{}(std::this_thread::get_id());
-    auto ctx = contexts[tid];
-    handleObservation(ctx);
-    cudaq::getExecutionManager()->resetExecutionContext();
-    contexts[tid] = nullptr;
-    contexts.erase(tid);
-  }
-};
+  bool supports_task_distribution() const override { return true; }
 
-class MultiQPUQuantumPlatform : public cudaq::quantum_platform {
-public:
-  ~MultiQPUQuantumPlatform() = default;
-  MultiQPUQuantumPlatform() {
-    int nDevices;
-    cudaGetDeviceCount(&nDevices);
-
-    auto envVal = spdlog::details::os::getenv("CUDAQ_MQPU_NGPUS");
-    if (!envVal.empty()) {
-      int specifiedNDevices = 0;
-      try {
-        specifiedNDevices = std::stoi(envVal);
-      } catch (...) {
+  void setTargetBackend(const std::string &description) override {
+    const auto getOpt = [](const std::string &str,
+                           const std::string &prefix) -> std::string {
+      const auto prefixPos = str.find(prefix);
+      if (prefixPos == std::string::npos)
+        return "";
+      const auto endPos = str.find_first_of(";", prefixPos);
+      return (endPos == std::string::npos)
+                 ? str.substr(prefixPos + prefix.size() + 1)
+                 : cudaq::split(str.substr(prefixPos + prefix.size() + 1),
+                                ';')[0];
+    };
+
+    if (description.find("remote_execution") != std::string::npos) {
+      if (!cudaq::registry::isRegistered<cudaq::QPU>("RemoteSimulatorQPU"))
         throw std::runtime_error(
-            "Invalid CUDAQ_MQPU_NGPUS environment variable, must be integer.");
+            "Unable to retrieve RemoteSimulatorQPU implementation.");
+      auto urls = cudaq::split(getOpt(description, "url"), ',');
+      auto sims = cudaq::split(getOpt(description, "backend"), ',');
+      // If no URL is provided, default to auto launching one server instance.
+      const bool autoLaunch =
+          description.find("auto_launch") != std::string::npos || urls.empty();
+
+      const auto formatUrl = [](const std::string &url) -> std::string {
+        auto formatted = url;
+        if (formatted.rfind("http", 0) != 0)
+          formatted = std::string("http://") + formatted;
+        if (formatted.back() != '/')
+          formatted += '/';
+        return formatted;
+      };
+
+      if (autoLaunch) {
+        urls.clear();
+        if (sims.empty())
+          sims.emplace_back("qpp");
+        const auto numInstanceStr = getOpt(description, "auto_launch");
+        // Default to launching one instance if no other setting is available.
+        const int numInstances =
+            numInstanceStr.empty() ? 1 : std::stoi(numInstanceStr);
+        cudaq::info("Auto launch {} REST servers", numInstances);
+        for (int i = 0; i < numInstances; ++i) {
+          m_remoteServers.emplace_back(
+              new cudaq::AutoLaunchRestServerProcess());
```

Comment:
Any reason that we can't use `unique_ptr` here?

Reply:
Fixed in https://github.com/NVIDIA/cuda-quantum/pull/1012/commits/c9076938f6face6bad1a09b0ab42f2e6f3f1b9cc

I imagine that I was just typing too fast, dropping the explicit constructor to rely on the in-place constructor :) Agree that we should make it more explicit.